### ISSUE TITLE
validationCallback doesn't respect timeout configuration: Possible deadlock

### ISSUE THREAD

**Issue by Noctunus** [citrusframework/citrus#778]
**Citrus Version**
2.8.0

**Expected behavior**
If a validationCallback is used while trying to receive messages the set timeout should be respected.

**Actual behavior**
If a httpserver is receiving a request but does not answer the test will just hang indefinitely. This happend for us while requesting a software where only the processing part crashed while processing the request. Therefore the http-connection was still active but no response got back to Citrus resulting in a complete hanging test.

**Test case sample**
Attached you'll find 2 files - first the .java file which is executed by Citrus and is using a very simple HttpServer-Mock written in Python3. As you can see the timeout in the test is configured for very low threshold and the HttpServer-Mock has a wait-time (simulating the part of not getting a response) of 10s before the Python HttpServer automatically sends a response. You can increase the wait time in the httpserver and you'll notice that the Citrus test is just hanging for the entire time the httpserver is not responding instead of running into a timeout.

[SSO3502_httpserver.py.txt](https://github.com/citrusframework/citrus/files/6625318/SSO3502_httpserver.py.txt)
[SSO3502_ValidationCallback_Deadlock.java.txt](https://github.com/citrusframework/citrus/files/6625319/SSO3502_ValidationCallback_Deadlock.java.txt)

**Comment by christophd** [comment]
yes, this is because Citrus only provides the endpoint component using Spring's RestTemplate and the HttpClient that is used behind the scenes will wait synchronously for the response with a longer timeout (maybe waiting forever as long as the connection is active). 

In case you want to set a custom request timeout please configure the underlying HttpClient implementation on the Spring RestTemplate (e.g. org.apache.httpcomponents:httpclient). You can set a custom client implementation or the Spring RestTemplate on the Citrus http-client endpoint component.

As an alternative you could also use the `fork=true` option when sending the request with Citrus. The corresponding `receive` test action on the same http-client will raise the timeout error with the configured timeout on the Citrus endpoint component then.

Hope that helps.
